src/parser.ts

```diff
@@ -40,71 +190,48 @@ export interface CodeSample {
 
 export type MimeType = 'application/json' | 'text/plain; charset=utf-8' | 'application/x-binary'
 
-export interface Schema {
-  description: string;
-  type?: XtpSchemaType;
-  enum?: string[];
-  properties?: { [name: string]: Property };
-  additionalProperties?: Property;
+export interface Schema extends XtpTyped {
+  description?: string;
   required?: string[];
+  properties?: { [name: string]: Property };
 }
 
-export type XtpSchemaType = 'object' | 'enum' | 'map'
+// TODO this figure out how to split up type again?
+//export type XtpSchemaType = 'object' | 'enum' | 'map'
+
 export type XtpType =
-  'integer' | 'string' | 'number' | 'boolean' | 'object' | 'array' | 'buffer';
+  'integer' | 'string' | 'number' | 'boolean' | 'object' |
+  'array' | 'buffer' | 'object' | 'enum' | 'map';
 export type XtpFormat =
   'int32' | 'int64' | 'float' | 'double' | 'date-time' | 'byte';
 
-export interface XtpItemType {
```

Comment:
This is really just another XtpTyped. items and addtionalProperties are no different than a property, parameter, etc